### PR TITLE
OPS-2376 Generalize repository structure

### PR DESCRIPTION
**Pull request by cytopia**
### Upcoming git tag: `v1.0.0`

### CODE REVIEW

File: tasks/_facts_route_nat_by_filter.yml [hunk 1]
```diff
@@ -8,14 +8,21 @@
 # Returns:
 # * aws_vpc_rt_nat_by_filter
 
+- name: combine nat filter
+  set_fact:
+    _aws_vpc_route_table_nat_filter: |
+      {{
+        aws_vpc_route_table_nat_filter_additional + nat.filter
```
Comment (danvaida) [inline tasks/_facts_route_nat_by_filter.yml]: nice one to extend the filter list ;)

File: tasks/_facts_route_vgw_by_filter.yml [hunk 2]
```diff
@@ -9,14 +9,21 @@
 # * aws_vpc_rt_vgw_by_filter
 # * aws_vpc_rt_vgw_propagate_by_filter
 
+- name: combine vgw filter
+  set_fact:
+    _aws_vpc_route_table_vgw_filter: |
+      {{
+        aws_vpc_route_table_vgw_filter_additional + vgw.filter
```
Comment (danvaida) [inline tasks/_facts_route_vgw_by_filter.yml]: for the sake of clarity, could you please explain why there's a need to have an additional filter?

File: tasks/create_vpc_route_table.yml [hunk 3]
```diff
@@ -18,13 +20,20 @@
       # type must be present
       - item.type is defined
       # type must be one of the following
-      - item.type in ['igw', 'nat', 'vgw', 'instance', 'interface', 'peering']
+      - item.type in ['igw', 'nat', 'vgw', 'instance', 'eni', 'peering']
       # igw does not require a 'filter' or 'name', the rest do
-      - (item.type in ['igw']) or (item.type in ['nat', 'vgw', 'instance', 'interface', 'peering'] and (item.filter is defined or item.name is defined))
+      - >-
+        (item.type in ['igw']) or
```
Comment (danvaida) [inline tasks/create_vpc_route_table.yml]: why not using the equal operand here, for one match only?

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+### Default
+###
+help:
+	@printf "%s\n\n" "Available commands"
+	@printf "%s\n"   "make test             Test the Ansible role"
+	@printf "%s\n"   "make lint             Lint source files"
+	@printf "%s\n"   "make help             Show help"
+
+test:
+	docker run --rm -it \
```
Comment (danvaida) [inline Makefile]: given that this is an AWS role, we'll need to also expose the credentials (preferably as env vars).

File: tasks/create_vpc_route_table.yml (hunk 3 above)
Comment (cytopia) [reply tasks/create_vpc_route_table.yml]: This is actually true ;p

File: tasks/_facts_route_vgw_by_filter.yml (hunk 2 above)
Comment (cytopia) [reply tasks/_facts_route_vgw_by_filter.yml]: Might become more clear once you have a look at the kubernetes PR

**Pull request merged**